### DISCUSSION TITLE
guide/features/content-search/

### DISCUSSION THREAD

**Discussion by giscus** [pengzhanbo/vuepress-theme-plume#118] [docs-comment]
# guide/features/content-search/

主题提供了两种方式支持 内容搜索。 本地内容搜索 Algolia DocSearch 注意，请勿同时配置使用两种方式，同时配置时，只有 本地内容搜索 生效。 本地内容搜索 本地内容搜索由 @vuepress-plume/plugin-search 插件提供支持。 该插件使用 minisearch 进行内容搜索。 启用 主题默认 启用 本地内容搜索 功能...

https://plume.pengzhanbo.cn/guide/features/content-search/

**Comment by Jasen6198** [comment]
本地内容搜索功能还是默认启用的吗？好像搜不到😂

**Reply by pengzhanbo** [reply]
是的，默认启用的。

**Comment by dingyuqi** [comment]
配置项中的文档连接点击后是404

**Reply by dingyuqi** [reply]
该页面有几个外部链接都已失效

**Reply by pengzhanbo** [reply]
链接已修复。

**Comment by MrHaoH** [comment]
大佬，怎么在自定义组件中使用主题内部的VPNavBarSearch组件？

**Reply by pengzhanbo** [reply]
`import VPNavBarSearch from 'vuepress-theme-plume/components/Nav/VPNavBarSearch.vue'` 或者
`import VPNavBarSearch from '@theme/Nav/VPNavBarSearch.vue'`

**Comment by zhengchen-zzZ** [comment]
在进行搜索的时候，用键盘光标选择然后按enter键无法跳转

**Reply by pengzhanbo** [reply]
已排查到问题，在下个版本进行修复。

**Comment by dlh-lj** [comment]
大佬。为什么有的内容搜索不到，或者是搜索的不全

**Reply by pengzhanbo** [reply]
本地搜索插件并没有对文章的全部内容建立索引，因为这样做形成的索引文件太大了。本地搜索插件主要是对文章的各级标题来形成索引以支持搜索功能。 如果你需要更强大更全面的搜索能力，推荐使用主题内置的 [algolia-docsearch](https://theme-plume.vuejs.press/guide/features/content-search/#algolia-docsearch)

**Comment by aiwandiannaodelele** [comment]
怎么关闭搜索？

**Reply by pengzhanbo** [reply]
https://theme-plume.vuejs.press/config/theme/#search
search 配置为 false 即可

**Comment by Soltus** [comment]
```ts
slimsearchPlugin({
      indexContent: true,
      suggestion: true,
      customFields: [
        {
          name: 'author',
          //@ts-ignore
          getter: (page) => page.frontmatter.author,
          formatter: '作者：$content',
        },]
    }),
```

内置的搜索可以像 slimsearchPlugin 这样定义检索 `customFields` 吗，没找到配置
